tests/test_auth.py

```diff
@@ -617,6 +617,38 @@ def test_refresh_aadtokencredentials_common_session(user_password):
         response = session.get("https://management.azure.com/subscriptions?api-version=2016-06-01")
         response.raise_for_status() # Should never raise
 
+@pytest.mark.slow
+def test_refresh_aadtokencredentials_existing_session(user_password):
```

Comment:
This test is actually the same as `test_refresh_aadtokencredentials_common_session`, since the returned session is the one in parameter:
https://github.com/Azure/msrest-for-python/blob/e465312c37a2887b773770e7c96a46a4fa86877f/msrest/authentication.py#L160-L173

Does this test really fail if your fix is not there?

Reply:
It sure does!  The difference is the test doesn't expect another session being returned like in `test_refresh_aadtokencredentials_common_session`.  It uses the same session which was originally being overwritten in `refresh_session` when it opened the context manager.  So before the fix it'd never be modified with the new token.

I need to update the test though, I make a mistake with the session variable in there.

Reply:
Ok, I get the full picture now. that's twisted bug :p 
Thank you!